tasks/03_create_key.yml

```diff
@@ -0,0 +1,40 @@
+---
+# In this Play an ssh key pair is created for login to the backup server and secure data transfer.
+# We do not want the possibly existing key to be used/distributed, 
+# so a backup-specific (backup/backup.pub) one is created.
+- name: Create ssh-key (if neeeded) for {{ borgbackup_user }} 
+  tags:
+    - install_backup
+  block:
+    - name: Ensire directory exist
+      ansible.builtin.file:
+        path: "{{ backup_user_info.home }}/.ssh/"
+        state: directory
+        mode: "0700"
+        owner: "{{ borgbackup_user }}"
+        group: "{{ borgbackup_group }}"
+
+    - name: Generate an OpenSSH keypair
+      community.crypto.openssh_keypair:
+        path: "{{ backup_user_info.home }}/.ssh/backup"
```

Comment:
The reasoning behind this is: With the restructed role it will be possible to implement the backup process under root or via a service user. If the user uses this role and decides to use the root acount, for security reasons I want the key to be distributed only to the systems where the backup takes place. 
I know few users who do key management on their SSH keys. Mostly keys are created only once and then used for years. In a complex environment the user tends to use public and private key of the environment wide root account.

I just say that inject via cloudinit or just ansible.
Since the role creates a key pair only when none exists....

Reply:
I'm not happy with this key stuff yet.

- `.ssh/backup` is hard-coded in the playbook, but also a variable - `borg_ssh_key_file`
- Too opinionated to force such a setup. Maybe I already use my existing key.
- If I run Borg commands manually, I need to take it into account.
- Unclear what happens on existing installs. I guess they will get the extra key

Reply:
I'm at a point now where I'm considering splitting off this Role. Not because your arguments are not conclusive and the approach to implement "always backwards compatible" is not right, but _my_ scope was to develop the role further.
A role should, from my point of view, be designed like a library.
By this I mean that the user should use this role to get a working backup without having to dig deep into the product, at the beginning. The second goal must be that the developers make sure that the role implements best practices and security first designs.

You are focused on making sure the role is "always backwards compatible".
Here I agree, however the user still only gets the security implementations I want to bring in if they know what they are doing and explicitly enable them.

My goals were:
  - to implement a service user that is only allowed to start borgmatic as root.
  - To minimize the foodprint, a ssh keys suitable for the purpose and no general keys.
  - End to end automation and tools for expected usage scenarios.

I like to finish restructuring with you. However, that is your role. If you don't share these goals, then we can reverse engineer a good portion of my changes.

To your points/questions:
  - Thanks for paying attention, keeping track in the various iterations of changes is difficult, I have changed everything to `borg_ssh_key_file`.
  - in context: made "always backwards compatible" correct and customizable
  - in context: "always backward compatible" correct, see above, is now customizable and the behavior is as before

Reply:
Thanks. I can review and make the final changes without raising more comments. Almost there, really.

You should have done the restructuring first. Using just the same tasks, but in different files. Then add the other improvements as separate PRs. Now we have one massive PR that's harder to review and discuss.